modules/effects/arrow-effects-rx2/src/test/kotlin/arrow/effects/SingleKTests.kt

```diff
@@ -0,0 +1,109 @@
+    override fun SingleKOf<T>.eqv(b: SingleKOf<T>): Boolean =
+      try {
+        this.value().blockingGet() == b.value().blockingGet()
+      } catch (throwable: Throwable) {
+        val errA = try {
+          this.value().blockingGet()
+          throw IllegalArgumentException()
+        } catch (err: Throwable) {
+          err
+        }
+        val errB = try {
+          b.value().blockingGet()
+          throw IllegalStateException()
+        } catch (err: Throwable) {
+          err
+        }
+        errA == errB
+      }
+
+  }
+
+  init {
+    testLaws(
+      FunctorLaws.laws(SingleK.functor(), { SingleK.just(it) }, EQ()),
+      ApplicativeLaws.laws(SingleK.applicative(), EQ()),
+      MonadLaws.laws(SingleK.monad(), EQ()),
+      MonadErrorLaws.laws(SingleK.monadError(), EQ(), EQ(), EQ()),
+      ApplicativeErrorLaws.laws(SingleK.applicativeError(), EQ(), EQ(), EQ()),
+      MonadSuspendLaws.laws(SingleK.monadDefer(), EQ(), EQ(), EQ()),
+      AsyncLaws.laws(SingleK.async(), EQ(), EQ(), EQ()),
+      AsyncLaws.laws(SingleK.effect(), EQ(), EQ(), EQ())
+    )
+
+    "Multi-thread Singles finish correctly" {
+      val value: Single<Long> = SingleK.monadError().bindingCatch {
+        val a = Single.timer(2, TimeUnit.SECONDS).k().bind()
+        a
+      }.value()
+
+      val test: TestObserver<Long> = value.test()
+      test.awaitDone(5, TimeUnit.SECONDS)
+      test.assertTerminated().assertComplete().assertNoErrors().assertValue(0)
+    }
+
+    "Multi-thread Observables should run on their required threads" {
+      val originalThread: Thread = Thread.currentThread()
+      var threadRef: Thread? = null
+
+      val value: Single<Long> = SingleK.monadError().bindingCatch {
+        val a = Single.timer(2, TimeUnit.SECONDS, Schedulers.newThread()).k().bind()
+        threadRef = Thread.currentThread()
+        val b = Single.just(a).observeOn(Schedulers.newThread()).k().bind()
+        b
+      }.value()
+
+      val test: TestObserver<Long> = value.test()
+      val lastThread: Thread = test.awaitDone(5, TimeUnit.SECONDS).lastThread()
+      val nextThread = (threadRef?.name ?: "")
+
+      nextThread shouldNotBeElseLogged originalThread.name
+      lastThread.name shouldNotBeElseLogged originalThread.name
+      lastThread.name shouldNotBeElseLogged nextThread
+    }
+
+    "Single dispose forces binding to cancel without completing too" {
+      val value: Single<Long> = SingleK.monadError().bindingCatch {
+        val a = Single.timer(3, TimeUnit.SECONDS).k().bind()
+        a
+      }.value()
+
+      val test: TestObserver<Long> = value.doOnSubscribe { subscription ->
+        Single.timer(1, TimeUnit.SECONDS).subscribe { a ->
+          subscription.dispose()
+        }
+      }.test()
+
+      test.awaitTerminalEvent(5, TimeUnit.SECONDS)
+      test.assertNotTerminated().assertNotComplete().assertNoErrors().assertNoValues()
+    }
+  }
+
+  // FIXME(paco): remove if this hasn't triggered in a while - 26 Jan 18
```

Comment:
lol is this in both Observable and Flowable? I wouldn't have it on by default.

Reply:
Yeah It's copypasta